api/db/migrations/20210607142018_fix-certifications-version-v2.js

```diff
@@ -0,0 +1,14 @@
+const TABLE_NAME = 'certification-courses';
+
+exports.up = function(knex) {
+  return knex(TABLE_NAME)
+    .where('createdAt', '>', '2021-01-01 00:00:00.000000')
```

Comment:
écrire simplement 'createdAt', '>', '2021-01-01' et il caste tout seul comme un grand 😄

Reply:
On a essayé la requête en prod ? Car cette requête là ne va pas être très efficace

Reply:
Ok je viens de tester, sur la base de réplication elle est assez longue, ~8 secondes.
Contrairement à ce que j'ai suggéré au début pour cette migration, j'ai testé en ajoutant dans le WHERE la clause "isV2Certification" IS FALSE, et ça va beaucoup beaucoup plus vite. Le plan est le même, mais du coup il fait beaucoup moins d'écritures en retenant exclusivement celles qu'il faut, et je n'avais pas pensé à ça.
Première requête : nombre de lignes réécrites -> ~410k, durée : ~8 secondes.
Deuxième requête : nombre de lignes réécrites -> 386, durée : ~200 ms.

Conclusion, il faut spécifier qu'on veut update seulement pour les certifs où isV2Certif est false.

Encore une fois, désolée pour la mauvaise recommandation

Reply:
Ah oui, c'est malin!

Reply:
@laura-bergoens J'ai ajouté la condition dans la migration, mais pour tester en prod, est-ce que tu réécris la requête en SQL pour la jouer sur la réplication ou est-ce qu'il y a moyen de tester la réplication elle-même ?

Reply:
Et sinon vous êtes sûrs que vous voulez utiliser une migration de DB pour ça ? Perso j'aurais fait une opération manuelle one-shot, vu qu'il n'y a aucun changement de schéma associé…

Reply:
Au-delà des questions de goût, il y a un impact concret à mettre ça dans une migration, c'est le risque que l'ancien code (qui tourne toujours pendant, et juste après, la migration) re-modifie incorrectement une des certifs qui aurait été corrigée par la migration.
Dans un cas comme ça ce qu'on veut plutôt c'est s'assurer que la correction du bug soit complètement déployée avant de rectifier les erreurs historiques.

Reply:
je me suis dit que c'était la meilleure solution pour corriger toutes les BDDs (intég et recette au passage aussi).
On a hésité avec le one-shot

Reply:
Si on a doute, rien n'empêche de faire une opération manuelle one-shot ensuite et de garder la migration pour s'assurer de corriger toutes les BDDs ?

Reply:
Du coup j'ai testé avec 
```sql
update "certification-courses" set "isV2Certification" = true where "createdAt" > '2021-01-01' and "isV2Certification" = false
```
Et la requête a mis 11 secondes pour 386 lignes. J'ai raté un truc ?

Reply:
Je pense que @laura-bergoens a raison de dire que la requête qui modifie moins de lignes va aller plus vite, en revanche c'est possible qu'en testant l'une juste après l'autre la différence de temps s'explique surtout par la mise en cache effectuée par la première…